sys/arduino/base.cpp

```diff
@@ -58,3 +63,40 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+/*
+ * Bitfield for the state of the ADC-channels.
+ * 0: Not initialized
+ * 1: Successfully initialized
+ */
+ static uint16_t adc_line_state = 0;
+
+int analogRead(int arduino_pin)
+{
+    int adc_value;
+
+    /* Check if the ADC line is valid */
+    assert((arduino_pin >= 0) && (arduino_pin < ANALOG_PIN_NUMOF));
+
+    /* Initialization of given ADC channel */
+    if (!(adc_line_state & (1 << arduino_pin))) {
+        if (adc_init(ADC_LINE(arduino_analog_map[arduino_pin])) == 0) {
+            /* The ADC channel is initialized */
+            adc_line_state |= 1 << arduino_pin;
+        }
+        else {
+            return -1;
+        }
+    }
+
+    /* Read the ADC channel */
+    adc_value = adc_sample(ADC_LINE(arduino_analog_map[arduino_pin]), ADC_RES_10BIT);
+
+    /* Check the result of ADC */
+    if (adc_value == -2) {
+        /* The resolution is not valid */
+        return -2;
```

Comment:
this `if` is not needed, as you return `adc_value` unchanged anyway.

Reply:
Yes, you are right ;)